src/openforms/js/components/admin/form_design/FormAdvancedConfiguration.js

```diff
@@ -0,0 +1,108 @@
+  const hasCosign = components.some(comp => comp.type === COSIGN_OLD_TYPE_KEY);
+
+  return (
+    <>
+      {hasBRPPrefill || hasNpFamilyMembers || hasCosign ? (
+        <Fieldset
+          title={
+            <FormattedMessage
+              defaultMessage="Processing"
+              description="Form processing fieldset title"
+            />
+          }
+        >
+          <FormRow>
+            <Field
+              name="form.brpPersonenRequestOptions.brpPersonenPurposeLimitationHeaderValue"
+              label={
+                <FormattedMessage
+                  defaultMessage="BRP Personen purpose limitation header value"
+                  description="Form 'BRP Personen purpose limitation header value' field label"
+                />
+              }
+              helpText={
+                <FormattedMessage
+                  defaultMessage='The purpose limitation ("doelbinding") for queries to the BRP Persoon API.'
+                  description="Form 'BRP Personen purpose limitation header value' field help text"
+                />
+              }
+            >
+              <TextInput
+                value={brpPersonenRequestOptions.brpPersonenPurposeLimitationHeaderValue}
+                onChange={onChange}
+                maxLength="255"
+              />
+            </Field>
+          </FormRow>
+          <FormRow>
+            <Field
+              name="form.brpPersonenRequestOptions.brpPersonenProcessingHeaderValue"
+              label={
+                <FormattedMessage
+                  defaultMessage="BRP Personen processing header value"
+                  description="Form 'BRP Personen processing header value' field label"
+                />
+              }
+              helpText={
+                <FormattedMessage
+                  defaultMessage='The processing ("verwerking") for queries to the BRP Persoon API.'
+                  description="Form 'BRP Personen processing header value' field help text"
+                />
+              }
+            >
+              <TextInput
+                value={brpPersonenRequestOptions.brpPersonenProcessingHeaderValue}
+                onChange={onChange}
+                maxLength="242"
+              />
+            </Field>
+          </FormRow>
+        </Fieldset>
+      ) : (
+        <p style={{padding: '1em 0 0', color: 'var(--body-quiet-color)'}}>
+          <FormattedMessage
+            defaultMessage="No advanced configuration is available for this form."
+            description="Form advanced configuration fallback message"
+          />
+        </p>
+      )}
+    </>
+  );
+};
+
+FormAdvancedConfiguration.propTypes = {
+  form: PropTypes.shape({
+    brpPersonenRequestOptions: PropTypes.shape({
+      brpPersonenPurposeLimitationHeaderValue: PropTypes.string,
+      brpPersonenProcessingHeaderValue: PropTypes.string,
+    }),
+  }).isRequired,
+  formSteps: PropTypes.array.isRequired,
```

Comment:
```suggestion
  formSteps: PropTypes.arrayOf(TYPES.FormStep).isRequired,
```
with https://github.com/open-formulieren/open-forms/blob/master/src/openforms/js/components/admin/form_design/types/index.js#L9, I think?